midend/local_copyprop.cpp

```diff
@@ -140,9 +154,20 @@ IR::AssignmentStatement *P4::DoLocalCopyPropagation::postorder(IR::AssignmentSta
     return as;
 }
 
+IR::MethodCallExpression *P4::DoLocalCopyPropagation::postorder(IR::MethodCallExpression *mc) {
+    if (!working) return mc;
+    LOG3("method call " << mc->method << "clears all saved values");
+    /* FIXME -- should look at what the called method actually reads and modifies, rather
+     * than assuming everything */
+    for (auto &var : Values(available)) {
```

Comment:
The way the def_use pass handles inter-procedural analysis is similar to the ControlFlowVisitor: it just allocates a new copy of itself sharing by reference the same state as 'this', and it uses it to visit the action/table body. Visiting a table is very similar to a switch statement - one branch for each action in the actions list.